tests/cxx/template_args.cc

```diff
@@ -115,6 +119,37 @@ void NestedTemplateArguments() {
   Outer<decltype(StaticTemplateFieldStruct::tpl)>* opsi;
   // IWYU: IndirectClass is...*indirect.h
   (void)opsi->t;
+
+  // Test obtaining template argument through alias template, typedef and
```

Comment:
This test sequence is pretty hard to read as there's so many different mechanisms going on (templates, `decltype` for sugar, providing vs. non-providing template definitions, pointers vs. not), _BUT_ they're ultimately very clear in contrasting the impact of the different mechanisms. Nicely done!